cli/cliui/agent.go

```diff
@@ -346,3 +349,42 @@ func PeerDiagnostics(w io.Writer, d tailnet.PeerDiagnostics) {
 		_, _ = fmt.Fprint(w, "✘ Wireguard is not connected\n")
 	}
 }
+
+type ConnDiags struct {
+	Info               *workspacesdk.AgentConnectionInfo
+	PingP2P            bool
+	LocalNetInfo       *tailcfg.NetInfo
+	FetchAgentNetcheck func() (healthsdk.AgentNetcheckReport, error)
+	// TODO: More diagnostics
+}
+
+func ConnDiagnostics(w io.Writer, d ConnDiags) {
+	if d.PingP2P {
+		_, _ = fmt.Fprint(w, "✔ You are connected directly, peer-to-peer (p2p).\n")
+		return
+	}
+	_, _ = fmt.Fprint(w, "❗ You are connected via a DERP relay, not directly, peer-to-peer (p2p).\n")
+
+	if d.Info != nil && d.Info.DisableDirectConnections {
+		_, _ = fmt.Fprint(w, "❗ Your Coder administrator has blocked direct connections.\n")
+		return
+	}
+
+	if d.Info != nil && d.Info.DERPMap != nil && !d.Info.DERPMap.HasSTUN() {
+		_, _ = fmt.Fprint(w, "✘ The workspace agent appears to be unable to reach any STUN servers.\nhttps://coder.com/docs/networking/stun")
+	}
+
+	if d.LocalNetInfo != nil && d.LocalNetInfo.MappingVariesByDestIP.EqualBool(true) {
+		_, _ = fmt.Fprint(w, "❗ Client is potentially behind a hard NAT, as multiple endpoints were retrieved from different STUN servers.\n")
+	}
+
+	agentReport, err := d.FetchAgentNetcheck()
+	if err != nil {
+		_, _ = fmt.Fprintf(w, "Failed to retrieve netcheck report from agent: %v\n", err)
```

Comment:
Old agents aren't going to have this API endpoint, so if it's a 404 you should ignore it or say something along the lines of "Your agent is outdated and a full report could not be generated"

Reply:
Accidentally removed this, will add it back